FrameworkSystem/Service/ProxyManagerHandler.py

```diff
@@ -84,13 +84,15 @@ def export_getUserProxiesInfo(self):
 
   types_requestDelegationUpload = [[int, long]]
 
-  def export_requestDelegationUpload(self, requestedUploadTime):
+  def export_requestDelegationUpload(self, requestedUploadTime, diracGroup=None):
     """ Request a delegation. Send a delegation request to client
 
         :param int,long requestedUploadTime: requested live time
 
         :return: S_OK(dict)/S_ERROR() -- dict contain id and proxy as string of the request
     """
+    if diracGroup:
+      return S_ERROR("Cannot upload proxy with DIRAC group or VOMS extensions; Please update your DIRAC client.")
```

Comment:
This is a completely breaking change. Running pilots and jobs will fail because of that. So this is probably not good.

Reply:
In v7 DIRAC group option not used in this method, all requests from old client will fail. But pilots and other "server side actions" will be from new DIRAC installation version, so its must work fine, how I understand.
This change for some old users Cli installation to notify "Client is old, need to update"

Reply:
all requests from old client will fail -> that is unacceptable. When we update version on teh servers, we still have many thousands of pilots and clients using the old version. There MUST be a backward compatibility of one version.

Reply:
The only thing that will fail is dirac-proxy-init from old clients. The user will receive a meaningful error message in this case suggesting upgrading the client. Old pilots are not affected. This is a compromise but a reasonable compromise I think. This should have been part of the major release (as was intended) v7r0 where we do not promise to keep interfaces compatible. Should we do v8r0 then ?

Reply:
Sorry my bad, I thought it was just requesting a delegation (also for downloading the proxy).
So indeed, it is only uploading a proxy that would fail. Moreover, the user would still get it locally available. So the issue is quite limited. But instead of returning an error, why not just ignoring it ? 
We do not keep the backward compatibility for major version ?! That's new to me. Up to which point can we break it ?

Reply:
Andrii is looking at ensuring backward compatibility also for the proxy upload from an old client. Looks like this is feasible. 
I did not say that we do not keep interface compatibility for new major release, but this is not guaranteed. Actually, changing interfaces is the main reason for a major release.

Reply:
It now seems compatible